### PR TITLE
Added --output_file flag

### PR DESCRIPTION
**Pull request by jmtsuji**
Adding the flag allows the output directory of the BLAST table to be specified by the user. This is needed for the snakemake workflow.

### CODE REVIEW

File: BackBLAST.py [hunk 1]
```diff
@@ -198,7 +199,7 @@ def main(args):
     if not subject_proteome_file.endswith(".faa"):
         print("[Warning] " + subject_proteome_file + " may not be a amino acid FASTA file!")
 
-    out_file = subject_proteome_file.rstrip(".faa") + ".csv"
+    # out_file = subject_proteome_file.rstrip(".faa") + ".csv"
```
Comment (LeeBergstrand) [inline BackBLAST.py]: There is no reason to leave commented out code if you are uses a source control system like Git. Having commented out code is considered bad for a variety of reasons. See: https://markhneedham.com/blog/2009/01/17/the-danger-of-commenting-out-code/

File: BackBLAST.py [hunk 2]
```diff
@@ -281,8 +282,11 @@ def main(args):
                              '''The smaller this number is, the stricter the BLAST search.''')
 
     parser.add_argument('-i', '--min_ident', metavar='IDENT', default=DEFAULT_MINIMUM_IDENTITY_CUTOFF,
-                        help='''The minimum sequence identify cutoff for removing high scoring pairs. ''' +
+                        help='''The minimum sequence identify cutoff for removing high scoring pairs.''' +
                              '''The larger this number is, the stricter the BLAST search.''')
 
+    parser.add_argument('-o', '--output_file', metavar='OUTPUT', required=True,
+                        help='''Output CSV-format BLAST results table.''')
```
Comment (LeeBergstrand) [inline BackBLAST.py]: Replace with something like "The path to the output file containing CSV-formatted BLAST results." Table is redundant. Indicate to the users that they are supposed to pass a file path.

**Comment by jmtsuji** [comment]
@LeeBergstrand Addressed -- thoughts?

**Comment by LeeBergstrand** [comment]
@jmtsuji You're free to merge.

**Pull request merged**